### PR TITLE
Fixed schema url in json dictionary

### PR DESCRIPTION
**Pull request by bnatrajan**
For standard schema files, schema url is now generated by scanning the un-versioned json file to find the url.
For annotation schema, value is hardcoded.
For standard schema files that don't have a corresponding json-schema, the closest url match is found.

### CODE REVIEW

File: rde_schema_dictionary_gen.py [hunk 1]
```diff
@@ -765,16 +791,100 @@ def fix_annotations_sequence_numbers(annotation_dictionary, annotation_index, st
...
+    elif ver32_ver1 < ver32_ver2:
+        return -1
+    else:
+        return 1
+
+
+def find_schema_url(json_schema_dirs, namespace, version, entity):
+    """
+    Returns the url for a specific namespace+version+entity
+
```
Comment (adkarira) [inline rde_schema_dictionary_gen.py]: whitespace

File: rde_schema_dictionary_gen.py [hunk 2]
```diff
@@ -765,16 +791,100 @@ def fix_annotations_sequence_numbers(annotation_dictionary, annotation_index, st
...
+def find_schema_url(json_schema_dirs, namespace, version, entity):
+    """
+    Returns the url for a specific namespace+version+entity
+	
+	Args:
+	    json_schema_dirs: list of json schema directories to search for url information
+		namespace: namespace the entity resides in
+		version: version information
+		entity: entity name
+
```
Comment (adkarira) [inline rde_schema_dictionary_gen.py]: whitespace

File: rde_schema_dictionary_gen.py [hunk 3]
```diff
@@ -765,16 +791,100 @@ def fix_annotations_sequence_numbers(annotation_dictionary, annotation_index, st
...
+    # in json-schema yet, we record the closest version found and return that url by modifying it's version.
+    closest_url = ''
+    closest_ver = ''
+
+    # find the un-versioned json schema file for this namespace
+    unversioned_schema_filename = namespace + '.json'
+    for json_schema_dir in json_schema_dirs:
+        if os.path.isfile(os.path.join(json_schema_dir, unversioned_schema_filename)):
+            with open(os.path.join(json_schema_dir, unversioned_schema_filename)) as file:
+                json_schema = json.load(file)
```
Comment (adkarira) [inline rde_schema_dictionary_gen.py]: json_schema should be available outside of the with statement.
We can move lines 856 - 878 4 spaces to the left. Nesting might look a little better.

File: rde_schema_dictionary_gen.py (hunk 2 above)
Comment (bnatrajan) [reply rde_schema_dictionary_gen.py]: fixed

File: rde_schema_dictionary_gen.py (hunk 3 above)
Comment (bnatrajan) [reply rde_schema_dictionary_gen.py]: yep, fixed.

**Pull request merged**